homeassistant/components/melcloud/climate.py

```diff
@@ -73,6 +86,18 @@
         True,
     )
 
+    platform = entity_platform.current_platform.get()
+    platform.async_register_entity_service(
+        SERVICE_SET_VANE_HORIZONTAL,
+        {vol.Required(CONF_POSITION): cv.string},
```

Comment:
After creating the attributes you can use them for the config validation e.g. vol.In(self._horizontal_positions)